torch_geometric/nn/conv/rgat_conv.py

```diff
@@ -0,0 +1,558 @@
+                torch.Tensor(self.num_bases, self.in_channels,
+                             self.heads * self.out_channels))
+        elif num_blocks is not None:
+            assert (self.in_channels % self.num_blocks == 0
+                    and (self.heads * self.out_channels) %
+                    self.num_blocks == 0), \
+                "both 'in_channels' and 'heads * out_channels' must be " \
+                "multiple of 'num_blocks' used"
+            self.weight = Parameter(
+                torch.Tensor(self.num_relations, self.num_blocks,
+                             self.in_channels // self.num_blocks,
+                             (self.heads * self.out_channels) //
+                             self.num_blocks))
+        else:
+            self.weight = Parameter(
+                torch.Tensor(self.num_relations, self.in_channels,
+                             self.heads * self.out_channels))
+
+        self.w = Parameter(torch.ones(self.out_channels))
+        self.l1 = Parameter(torch.FloatTensor(1, self.out_channels))
+        self.b1 = Parameter(torch.FloatTensor(1, self.out_channels))
+        self.l2 = Parameter(
+            torch.FloatTensor(self.out_channels, self.out_channels))
+        self.b2 = Parameter(torch.FloatTensor(1, self.out_channels))
+
+        self._alpha = None
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        if self.num_bases is not None:
+            glorot(self.basis)
+            glorot(self.att)
+        else:
+            glorot(self.weight)
+        glorot(self.q)
+        glorot(self.k)
+        zeros(self.bias)
+        ones(self.l1)
+        zeros(self.b1)
+        torch.full(self.l2.size(), 1 / self.out_channels)
+        zeros(self.b2)
+        if self.lin_edge is not None:
+            glorot(self.lin_edge)
+            glorot(self.e)
+
+    def forward(self, x: Tensor, edge_index: Adj, edge_type: OptTensor = None,
+                edge_attr: OptTensor = None, size: Size = None,
+                return_attention_weights=None):
+        r"""
+        Args:
+            x: The input node features. Can be either a :obj:`[num_nodes,
+                in_channels]` node feature matrix, or an optional
+                one-dimensional node index tensor (in which case input features
+                are treated as trainable node embeddings).
+            edge_type: The one-dimensional relation type/index for each edge in
+                :obj:`edge_index`.
+                Should be only :obj:`None` in case :obj:`edge_index` is of type
+                :class:`torch_sparse.tensor.SparseTensor`.
+                (default: :obj:`None`)
+            return_attention_weights (bool, optional): If set to :obj:`True`,
+                will additionally return the tuple
+                :obj:`(edge_index, attention_weights)`, holding the computed
+                attention weights for each edge. (default: :obj:`None`)
+        """
+
+        if self.add_self_loops:
```

Comment:
We should remove this argument here. Adding self-loops will throw errors later on since only `edge_attr` is modified but not `edge_type`.